projects/gateway2/proxy_syncer/endpoints_prioritize.go

```diff
@@ -48,14 +53,29 @@ func NewPriorities(failoverPriority []string) *Prioritizer {
 	}
 }
 
-func (p *Prioritizer) GetPriority(proxyLabels, upstreamEndpointLabels map[string]string) int {
+func (p *Prioritizer) GetPriority(proxyLabels, upstreamEndpointLabels map[string]string, defaultNetwork string) int {
 	for j, label := range p.priorityLabels {
 		valueForProxy, ok := p.priorityLabelOverrides[label]
 		if !ok {
-			valueForProxy = proxyLabels[label]
+			valueForProxy, ok = proxyLabels[label]
+			if !ok && label == istiolabel.TopologyNetwork.Name {
+				valueForProxy = defaultNetwork
+			}
+		}
+
+		valueForEndpoint, ok := upstreamEndpointLabels[label]
+		if !ok && label == istiolabel.TopologyNetwork.Name {
+			valueForEndpoint = defaultNetwork
+		}
+
+		valuesEqual := valueForProxy == valueForEndpoint
+
+		// istio network semantic: if either side is empty, assume flat network
+		if !valuesEqual && label == istiolabel.TopologyNetwork.Name {
```

Comment:
how will this condition be met? In the case where the lbael is the `istiolabel.TopologyNetwork.Name`, don't we set the valueForProxy and valueForEndpoint to the defaultNetwork?

Reply:
defaultNetwork can be empty

let's say I have Istio installed, but I haven't set the network for my local cluster yet; some controller writes a WorkloadEntry or Pod with `topology.istio.io/network=foo`. The precedent set in istio is to assume that endpoint is reachable even though `"" != "foo"`.